snapcraft/internal/project_loader/_config.py

```diff
@@ -19,33 +19,39 @@
 import os
 import os.path
 import re
+import urllib
+from typing import List, Set
 
 import jsonschema
-from typing import List, Set
 
-from snapcraft import plugins, project, formatting_utils
+from snapcraft import formatting_utils, plugins, project
 from snapcraft.internal import deprecations, repo, states, steps
 from snapcraft.internal.meta.snap import Snap
 from snapcraft.internal.pluginhandler._part_environment import (
     get_snapcraft_global_environment,
 )
 from snapcraft.project._schema import Validator
-from ._parts_config import PartsConfig
-from ._extensions import apply_extensions
-from ._env import build_env_for_stage, runtime_env, environment_to_replacements
-from . import errors, grammar_processing, replace_attr
 
+from . import errors, grammar_processing, replace_attr
+from ._env import build_env_for_stage, environment_to_replacements, runtime_env
+from ._extensions import apply_extensions
+from ._parts_config import PartsConfig
 
 logger = logging.getLogger(__name__)
 
 
 @jsonschema.FormatChecker.cls_checks("icon-path")
 def _validate_icon(instance):
+    parsed_url = urllib.parse.urlparse(instance)
```

Comment:
I would keep the generic implementation, but still disallow urls as icon entries in snapcraft.yaml

Reply:
Makes sense - I removed the urlparse bits and removed the icon path checking out of this schema validation, into conduct_project_sanity_check().  That way we still get the up-front checking, but doesn't require the schema to enforce such a check (which requires state outside of the yaml config).